subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/resolver/ExternalResourceResolver.java

```diff
@@ -325,7 +325,8 @@ private void publishPossiblyUnsupportedChecksum(ExternalResourceName destination
         try {
             publishChecksum(destination, content, algorithm, length);
         } catch (Exception ex) {
-            LOGGER.warn("Cannot upload checksum for " + content.getName() + ". Remote repository doesn't support " + algorithm + ". Error: " + ex.getMessage());
+            LOGGER.warn("Cannot upload checksum for " + content.getName() + " because the remote repository doesn't support " + algorithm + ". This will not fail the build.");
+            LOGGER.info("Upload error", ex);
```

Comment:
I do not like having two logging events emitted from a single location. They could end up well apart in the output.
Why don't you check if the lower level is enabled, then post the message with the exception _at the higher level anyway_?
And if the level is not enabled, post the message without the exception.
I also wonder if the exception is really useful in `info` and should not be limited to `debug`.

Reply:
Good call, applied your suggestion!